### PR TITLE
Don't use goroutine on loop iterator 🛑

### PR DESCRIPTION
**Pull request by bobcatfish**
# Changes

In some of our end to end tests, we have loops which create goroutines.
If you refer to loop iterator variables in those goroutines, you'll end
up using the same variable in all instances, since closures are bound to
only one variable.

For reference: https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables

Also removed some channels which were a bit more complicated than we
need (can't call Fatal in a goroutine, but Error is okay!)

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [tests](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if functionality changed/added)
- [n/a] Includes [docs](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if user facing)
- [n/a] Commit messages follow [commit message best practices](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
n/a
```

### CODE REVIEW

File: test/cancel_test.go [hunk 1]
```diff
@@ -82,29 +82,22 @@ func TestTaskRunPipelineRunCancel(t *testing.T) {
 	}
 
 	t.Logf("Waiting for TaskRuns from PipelineRun %s in namespace %s to be running", "pear", namespace)
-	errChan := make(chan error, len(taskrunList.Items))
-	defer close(errChan)
-
 	for _, taskrunItem := range taskrunList.Items {
-		go func(chan error) {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```
Comment (vdemeester) [inline test/cancel_test.go]:
```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop

File: test/cancel_test.go [hunk 2]
```diff
@@ -135,30 +128,25 @@ func TestTaskRunPipelineRunCancel(t *testing.T) {
 	}
 
 	t.Logf("Waiting for TaskRuns in PipelineRun %s in namespace %s to be cancelled", "pear", namespace)
-	errChan2 := make(chan error, len(taskrunList.Items))
-	defer close(errChan2)
 	for _, taskrunItem := range taskrunList.Items {
-		go func(errChan2 chan error) {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```
Comment (vdemeester) [inline test/cancel_test.go]: Same here

```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop

File: test/timeout_test.go [hunk 3]
```diff
@@ -84,20 +84,20 @@ func TestPipelineRunTimeout(t *testing.T) {
 	defer close(errChan)
 
 	for _, taskrunItem := range taskrunList.Items {
-		go func() {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```
Comment (vdemeester) [inline test/timeout_test.go]: and here 
```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop

File: test/timeout_test.go [hunk 4]
```diff
@@ -129,33 +129,26 @@ func TestPipelineRunTimeout(t *testing.T) {
 	}
 
 	t.Logf("Waiting for TaskRuns from PipelineRun %s in namespace %s to be cancelled", pipelineRun.Name, namespace)
-	errChan2 := make(chan error, len(taskrunList.Items))
-	defer close(errChan2)
-
 	for _, taskrunItem := range taskrunList.Items {
-		go func() {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```
Comment (vdemeester) [inline test/timeout_test.go]: and here :)

```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop

File: test/timeout_test.go (hunk 3 above)
Comment (vdemeester) [reply test/timeout_test.go]: oh wait we don't need for those ones I think

**Comment by bobcatfish** [comment]
Thanks for finishing this up @vdemeester !! :pray: :heart: :pray:

**Pull request merged**